netlogo-core/src/test/fileformat/NLogoFormatTests.scala

```diff
@@ -111,6 +111,42 @@ class NLogoFormatConversionTest extends FunSuite with ConversionHelper {
       assert(rereadModel.code.contains("extensions [vid]"))
     }
 
+    test("performs tests for sectionToSource conversion conversion between files") {
+      val sectionsMap = collection.immutable.Map("org.nlogo.modelsection.version" -> Array[String]("NetLogo 3D 6.0.4"))
+      val nlogoformat = new NLogoFormat
+      val expectedResult = 
+        """@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |NetLogo 6.0.4
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@""".trim.stripMargin
+      val expectedThreedResult = 
+        """@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |NetLogo 3D 6.0.4
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@
+          |@#$#@#$#@""".trim.stripMargin
+
+      assert(nlogoformat.sectionsToSource(sectionsMap).get.trim == expectedResult)
+      val nlogoformatThreed = new NLogoThreeDFormat
+      assert(nlogoformatThreed.sectionsToSource(sectionsMap).get.trim != expectedResult)
```

Comment:
It's jarring to see a test check that `someVal != expectedResult`, normally you want to see your expected result as the value.  Maybe `expectedResult` could be renamed to match `expectedThreedResult` so it's more clear why it should be different?